example/canny.cpp

```diff
@@ -0,0 +1,165 @@
+#include <boost/gil.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/numeric/sampler.hpp>
+#include <boost/gil/extension/numeric/resample.hpp>
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/extension/numeric/convolve.hpp>
+
+#include <math.h>
+
+using namespace std;
+using namespace boost::gil;
+
+//Example how to detect edges via Canny algorithm
+
+template <typename SrcView>
+void smoothing(const SrcView &src)
```

Comment:
`gaussian_blur` or `gaussian_smoothing` wouldn't be a clearer name?

It is an example with learning as primary purpose, isn't it.